tests/python/pants_test/java/test_nailgun_client.py

```diff
@@ -178,21 +178,3 @@ def test_execute_protocolerror_on_execute(self, mock_session, mock_try_connect):
 
   def test_repr(self):
     self.assertIsNotNone(repr(self.nailgun_client))
-
-  @mock.patch('os.kill', **PATCH_OPTS)
-  def test_send_control_c(self, mock_kill):
-    self.nailgun_client._session = mock.Mock(remote_pid=31337)
-    self.nailgun_client.send_control_c()
-    mock_kill.assert_called_once_with(31337, signal.SIGINT)
-
-  @mock.patch('os.kill', **PATCH_OPTS)
-  def test_send_control_c_noop_none(self, mock_kill):
-    self.nailgun_client._session = None
-    self.nailgun_client.send_control_c()
-    mock_kill.assert_not_called()
-
-  @mock.patch('os.kill', **PATCH_OPTS)
```

Comment:
This test seems useful... would be good to preserve if possible. If it was broken beforehand, leaving it in as `xfail` would be good, because there is a race between getting a pid from the server and `Ctrl+C` arriving at the client that means it's important to be able to noop.

Reply:
That race is erased in #6579 with the pid/pgrp being required before the rest of the communication continues, btw. I have added these back, fixed the mock so they pass, and added an additional test for the new `maybe_send_signal()` method.